src/credential-status/status-token.ts

```diff
@@ -0,0 +1,159 @@
+        aggregationUri: options.claimsSet.aggregationUri,
+      }),
+    }
+    if (options.claimsSet.expirationTime) {
+      claims[CwtStatusListClaims.ExpirationTime] = options.claimsSet.expirationTime
+    }
+    if (options.claimsSet.timeToLive) {
+      claims[CwtStatusListClaims.TimeToLive] = options.claimsSet.timeToLive
+    }
+
+    cwt.setClaims(claims)
+    return cborEncode(new Tag(await cwt.create({ type: options.type, key: options.key }), CoseTypeToTag[options.type]))
+  }
+
+  static async verifyStatusToken(options: CWTStatusTokenVerifyOptions): Promise<boolean> {
+    const cwt = cborDecode(options.token) as Sign1 | Mac0
+
+    const type = cwt.protectedHeaders.headers?.get(String(CwtProtectedHeaders.TYPE))
+    if (!type || type !== 'application/statuslist+cwt') {
+      throw new Error('CWT status token does not have the correct type in protected headers')
+    }
+
+    if (!cwt.payload) {
+      throw new Error('CWT status token does not contain claims')
+    }
+    const claims = cborDecode(cwt.payload) as Map<string, string | number | Uint8Array>
+    // Check if is the same as the one used to fetch the token
+    if (!claims.has(String(CwtStatusListClaims.StatusListUri))) {
+      throw new Error('CWT status token does not contain status list URI')
+    }
+    if (!claims.has(String(CwtStatusListClaims.IssuedAt))) {
+      throw new Error('CWT status token does not contain issued at claim')
+    }
+    if (!claims.has(String(CwtStatusListClaims.StatusList))) {
+      throw new Error('CWT status token does not contain status list')
+    }
+
+    const expirationTime = claims.get(String(CwtStatusListClaims.ExpirationTime))
+    if (expirationTime && typeof expirationTime === 'number' && expirationTime < Math.floor(Date.now() / 1000)) {
+      throw new Error('CWT status token has expired')
+    }
+
+    let coseType: CoseType
+    if (cwt instanceof Sign1) {
+      coseType = CoseType.Sign1
+    } else if (cwt instanceof Mac0) {
+      coseType = CoseType.Mac0
+    } else {
+      throw new Error('Unimplemented/Unsupported CWT type')
+    }
+    const validSignature = await CWT.verify({ type: coseType, token: options.token, key: options.key })
+    if (!validSignature) {
+      throw new Error('Invalid signature for CWT status token')
+    }
+
+    return true
+  }
+
+  static async verifyStatus(options: CWTStatusVerifyOptions): Promise<boolean> {
+    const validStatusToken = await CWTStatusToken.verifyStatusToken(options)
+    if (validStatusToken) {
+      const cwt = cborDecode(options.token) as Sign1 | Mac0
+      if (!cwt.payload) {
+        throw new Error('CWT status token does not contain claims')
+      }
+
+      const claims = cborDecode(cwt.payload) as Map<string, string | number | Uint8Array>
+      const statusList = claims.get(String(CwtStatusListClaims.StatusList))
+      return StatusList.verifyStatus(statusList as Uint8Array, options.index, options.expectedStatus)
+    }
```

Comment:
This is weird, The `verifyStatusToken` should return the parsed token, so we don't have to parse it again.


E.g. something like:

```ts
const statusTokenResult = await CWTStatusToken.verifyStatusToken(options)

if (statusTokenResult.isValid) {
  throw statusTokenResult.error
}

// isValid is now `true`
// use statusTokenResult.statusToken, which is typed

```

The return type of the `veirfyStatusToken` should then also be something like:

```
type VerifyStatusTokenResult = { isValid: true; statusToken: <todo: TypedStatusToken> }  | { isValid: false ; error: Error }
```